src/utils/getColumnWidth.ts

```diff
@@ -0,0 +1,46 @@
+export const MAX_COLUMN_WIDTH = 600;
+export const HEADER_PADDING = 20;
+export const SORT_ICON_PADDING = 18;
+export const PIXELS_PER_CHARACTER = 10;
+
+export function getColumnWidth({
+    data,
+    name,
+    header,
+    sortable,
+}: {
+    data?: Record<string, unknown>[];
+    name: string;
+    header?: string;
+    sortable?: boolean;
+}) {
+    const headerContentLength = typeof header === 'string' ? header.length : name.length;
+
+    let maxColumnContentLength = headerContentLength;
+
+    if (data) {
+        for (const row of data) {
+            let cellLength = 0;
+            if (row[name]) {
+                cellLength = String(row[name]).length;
+            }
+
+            maxColumnContentLength = Math.max(maxColumnContentLength, cellLength);
+
+            const sortPadding =
+                sortable && maxColumnContentLength <= headerContentLength ? SORT_ICON_PADDING : 0;
```

Comment:
There is a problem with such calculations. In the example your code will return 70, while it should return 78:
```js
maxColumnContentLength = 5
headerContentLength = 4
sortable = true

sortPadding = 0

actualColumnContentLength = 10 * 5 + 20 // 70
actualHeaderContentLength = 10 * 4 + 18 + 20 // 78
```

You may assume, that sort icon length is equal to 2 characters and just add this to `headerContentLength` - there will be 2px difference, you won't need to adjust you calculations for `sortPadding`
```js
const headerContentLength = typeof header === 'string' ? header.length : name.length;
const headerContentLengthWithSort = sortable ? headerContentLength + 2 : headerContentLength
```

Or you may compare actual widths, but not length in characters